ca_tools/scripts/draw_square.py

```diff
@@ -0,0 +1,209 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+
+
+import math
+from GTSys import GTSys
+from RobotLocalizationTf import RobotLocalizationTf
+
+import rospy
+from geometry_msgs.msg import Twist, Pose, Quaternion, PoseWithCovariance
+from nav_msgs.msg import Odometry
+from std_msgs.msg import Bool
+from tf.transformations import euler_from_quaternion as efq
+from tf.transformations import quaternion_from_euler as qfe
+
+
+
+class RobotState():
+
+    STATES = [
+    "MOVE_FORWARD",
+    "ROTATE",
+    "STOP"
+    ]
+
+    _current_state = None
+
+    def __init__(self):
+        self._current_state = "STOP"
+
+    def SetState(self,state):
+        "ToDo: assert a valid state"
+        self._current_state = state
+
+    def GetState(self):
+        return self._current_state
+
+
+
+
+
+class DrawSquare():
+
+    ANGLE_THRESHOLD = 0.05
+    ANGULAR_VEL = 1
+    LINEAR_VEL = 1
+    LENGTH_THRESHOLD = 0.05
+    GOAL_DIST_THRESHOLD = 0.1
+
+    _current_pose = Pose()
+    _state = RobotState()
+    _twist = Twist()
+    _next_pose = Pose()
+    _ground_truth = GTSys()
+    _goal_angle = 0
+    _currentGoalReached = False
+    _goal_pose = Pose()
+    _diff_angle = 0
+    _length_diff = 0
+    _square_length = 2
+    _tf = RobotLocalizationTf()
+
+
+    def __init__(self):
+        self._pub_cmd = rospy.Publisher('cmd_vel', Twist, queue_size=10)
+        self._pub_controller_on = rospy.Publisher('controller_on', Bool, queue_size=10)
+        self._pub_controller_goal = rospy.Publisher('goal_pose', Pose, queue_size=10)
+        self._hz = rospy.get_param('~hz', 10)
+        self._current_pose = self._ground_truth.get_pose()
+        self._rate = rospy.Rate(self._hz)
+        self._rate.sleep()
+
+    def _wrap_angle(self,angle): #ToDo: make beautiful this horrible function
```

Comment:
Isn't this function defined elsewhere? You might want to extract it as a library function shared among modules